adapters/repos/db/lsmkv/segment_serialization.go

```diff
@@ -352,32 +362,58 @@ func ParseCollectionNode(r io.Reader) (segmentCollectionNode, error) {
 // As a result calling this method only makes sense if you plan on calling it
 // multiple times. Calling it just once on an uninitialized node does not have
 // major advantages over calling ParseCollectionNode.
-func ParseCollectionNodeInto(in []byte, node *segmentCollectionNode) error {
+func ParseCollectionNodeInto(r io.Reader, node *segmentCollectionNode) error {
 	// offset is only the local offset relative to "in". In the end we need to
 	// update the global offset.
 	offset := 0
 
-	valuesLen := binary.LittleEndian.Uint64(in[offset : offset+8])
+	buf := make([]byte, 9)
+
+	_, err := io.ReadFull(r, buf[0:8])
+	if err != nil {
+		return fmt.Errorf("read values len: %w", err)
+	}
+
+	valuesLen := binary.LittleEndian.Uint64(buf[offset : offset+8])
 	offset += 8
 
 	resizeValuesOfCollectionNode(node, valuesLen)
 	for i := range node.values {
-		node.values[i].tombstone = in[offset] == 0x1
+		buf = make([]byte, 9)
```

Comment:
no need to allocate new bytes. previous slice can be reused to read new bytes into